### PR TITLE
Add timeouts to Merkle tree tests

### PR DESCRIPTION
**Pull request by wintersteiger**
This adjusts the numbers on Merkle tree tests so that they don't take too much time during CI runs and it adds timeout checking so no spurious failures are reported.

### CODE REVIEW

File: src/libmerklecpp/test/flush.cpp [hunk 1]
```diff
@@ -12,24 +12,27 @@
 
 int main()
 {
+  auto test_start_time = std::chrono::high_resolution_clock::now();
+  double timeout = get_timeout();
```
Comment (achamayou) [inline src/libmerklecpp/test/flush.cpp:16]: That's oddly specific but ok :)

**Review by achamayou — APPROVED** [review]
Thank you, that will prevent the tests from timing out in Daily while running completely in CI context.

File: src/libmerklecpp/test/flush.cpp (hunk 1 above)
Comment (wintersteiger) [reply src/libmerklecpp/test/flush.cpp:16]: How so? It gets the timeout from the environment variable.

Comment (achamayou) [reply src/libmerklecpp/test/flush.cpp:16]: I meant making it explicitly a double when everything around is auto :)

**Comment by achamayou** [comment]
This probably needs a force merge too since it fixes an issue in the Daily.

**Comment by wintersteiger** [comment]
Good point, yes. It threw a different error the last time, I'll check whether it still does so.

File: src/libmerklecpp/test/flush.cpp (hunk 1 above)
Comment (wintersteiger) [reply src/libmerklecpp/test/flush.cpp:16]: Oh yes, that's unnecessary :)

**Pull request merged**